### ISSUE TITLE
Various test case bugs

### ISSUE THREAD

**Issue by anba** [tc39/test262#1338]
The following test cases are incorrect on c4e3d12597fae7e56fa475122aefb7bee3a16f84:

```
# Accessing `ta[0]` throws a TypeError.
test/built-ins/TypedArrays/internals/DefineOwnProperty/tonumber-value-detached-buffer.js
test/built-ins/TypedArrays/internals/Set/tonumber-value-detached-buffer.js

# array indices start at 0
test/language/expressions/class/fields-asi-5.js
test/language/statements/class/fields-asi-5.js

# static property tested on instance instead of class
test/language/expressions/class/fields-computed-name-static-propname-constructor.js
test/language/expressions/class/fields-computed-name-static-propname-prototype.js
test/language/expressions/class/fields-computed-variable-name-static-propname-constructor.js
test/language/expressions/class/fields-computed-variable-name-static-propname-prototype.js
test/language/statements/class/fields-computed-name-static-propname-constructor.js
test/language/statements/class/fields-computed-name-static-propname-prototype.js
test/language/statements/class/fields-computed-variable-name-static-propname-constructor.js
test/language/statements/class/fields-computed-variable-name-static-propname-prototype.js

# `assert.equal` is not defined
test/language/module-code/privatename-valid-no-earlyerr.js
test/language/statements/class/privatename-valid-no-earlyerr.js

# `verifyProperty` used, but harness file not included
test/language/expressions/class/fields-after-same-line-gen-private-names.js
test/language/expressions/class/fields-after-same-line-gen-static-private-names.js
test/language/expressions/class/fields-after-same-line-method-private-names.js
test/language/expressions/class/fields-after-same-line-method-static-private-names.js
test/language/expressions/class/fields-after-same-line-static-async-gen-private-names.js
test/language/expressions/class/fields-after-same-line-static-async-gen-static-private-names.js
test/language/expressions/class/fields-after-same-line-static-async-method-private-names.js
test/language/expressions/class/fields-after-same-line-static-async-method-static-private-names.js
test/language/expressions/class/fields-after-same-line-static-gen-private-names.js
test/language/expressions/class/fields-after-same-line-static-gen-static-private-names.js
test/language/expressions/class/fields-after-same-line-static-method-private-names.js
test/language/expressions/class/fields-after-same-line-static-method-static-private-names.js
test/language/expressions/class/fields-multiple-definitions-private-names.js
test/language/expressions/class/fields-multiple-definitions-static-private-names.js
test/language/expressions/class/fields-multiple-stacked-definitions-private-names.js
test/language/expressions/class/fields-multiple-stacked-definitions-static-private-names.js
test/language/expressions/class/fields-new-no-sc-line-method-private-names.js
test/language/expressions/class/fields-new-no-sc-line-method-static-private-names.js
test/language/expressions/class/fields-new-sc-line-gen-private-names.js
test/language/expressions/class/fields-new-sc-line-gen-static-private-names.js
test/language/expressions/class/fields-new-sc-line-method-private-names.js
test/language/expressions/class/fields-new-sc-line-method-static-private-names.js
test/language/expressions/class/fields-same-line-async-gen-private-names.js
test/language/expressions/class/fields-same-line-async-gen-static-private-names.js
test/language/expressions/class/fields-same-line-async-method-private-names.js
test/language/expressions/class/fields-same-line-async-method-static-private-names.js
test/language/expressions/class/fields-same-line-gen-private-names.js
test/language/expressions/class/fields-same-line-gen-static-private-names.js
test/language/expressions/class/fields-same-line-method-private-names.js
test/language/expressions/class/fields-same-line-method-static-private-names.js
test/language/statements/class/fields-after-same-line-gen-private-names.js
test/language/statements/class/fields-after-same-line-gen-static-private-names.js
test/language/statements/class/fields-after-same-line-method-private-names.js
test/language/statements/class/fields-after-same-line-method-static-private-names.js
test/language/statements/class/fields-after-same-line-static-async-gen-private-names.js
test/language/statements/class/fields-after-same-line-static-async-gen-static-private-names.js
test/language/statements/class/fields-after-same-line-static-async-method-private-names.js
test/language/statements/class/fields-after-same-line-static-async-method-static-private-names.js
test/language/statements/class/fields-after-same-line-static-gen-private-names.js
test/language/statements/class/fields-after-same-line-static-gen-static-private-names.js
test/language/statements/class/fields-after-same-line-static-method-private-names.js
test/language/statements/class/fields-after-same-line-static-method-static-private-names.js
test/language/statements/class/fields-multiple-definitions-private-names.js
test/language/statements/class/fields-multiple-definitions-static-private-names.js
test/language/statements/class/fields-multiple-stacked-definitions-private-names.js
test/language/statements/class/fields-multiple-stacked-definitions-static-private-names.js
test/language/statements/class/fields-new-no-sc-line-method-private-names.js
test/language/statements/class/fields-new-no-sc-line-method-static-private-names.js
test/language/statements/class/fields-new-sc-line-gen-private-names.js
test/language/statements/class/fields-new-sc-line-gen-static-private-names.js
test/language/statements/class/fields-new-sc-line-method-private-names.js
test/language/statements/class/fields-new-sc-line-method-static-private-names.js
test/language/statements/class/fields-same-line-async-gen-private-names.js
test/language/statements/class/fields-same-line-async-gen-static-private-names.js
test/language/statements/class/fields-same-line-async-method-private-names.js
test/language/statements/class/fields-same-line-async-method-static-private-names.js
test/language/statements/class/fields-same-line-gen-private-names.js
test/language/statements/class/fields-same-line-gen-static-private-names.js
test/language/statements/class/fields-same-line-method-private-names.js
test/language/statements/class/fields-same-line-method-static-private-names.js

# typo s/Avalue/42/
test/language/statements/class/privatefieldget-success-5.js

# derived classes have access to private names in base classes, if private names are in scope
test/language/statements/class/privatefieldget-typeerror-3.js
test/language/statements/class/privatefieldset-typeerror-3.js

# missing "var" for strict mode
test/language/statements/class/privatefieldget-typeerror-4.js
test/language/statements/class/privatefieldset-typeerror-4.js
```

**Comment by littledan** [comment]
Cc @xanplz for claas field bugs

**Comment by leobalter** [comment]
@anba in #1339 I believe I'm fixing all of these bugs except for these:

```
# derived classes have access to private names in base classes, if private names are in scope
test/language/statements/class/privatefieldget-typeerror-3.js
test/language/statements/class/privatefieldset-typeerror-3.js
```

Which I believe the field from the outer class should not be accessed in the inner class through `this` even if they are nested.

I might be wrong, and I'm biased to say you're probably right. Even though, I'd like to confirm what is the intended behavior from @littledan and ask for a hint in the specs that I can confirm this, please.

cc @spectranaut.

**Comment by spectranaut** [comment]
@anba thanks for your work in catching catching all those typos! :)
 
```
# derived classes have access to private names in base classes, if private names are in scope
test/language/statements/class/privatefieldget-typeerror-3.js
test/language/statements/class/privatefieldset-typeerror-3.js
```
I believe these public fields will not be discovered because you do not walk up the prototype chain when looking for private fields, but I don't have more time to look now! Curious what @littledan will say.

**Comment by littledan** [comment]
I think I agree with @anba here, that no TypeError should be thrown. `#x` is in scope because it is lexically nested, and it is present on the receiver because `Inner` inherits from `Outer` and it uses the default constructor, so the superclass constructor will run and add `#x` to `this`. There's no need to walk up the prototype chain here: `#x` is placed on the instance under construction. Anyway, great test case to get at this edge case!

**Comment by leobalter** [comment]
Thanks for the feedback, @spectranaut and @littledan. I'll fix the tests. In the meanwhile, it feels so weird as this edge case allows nesting - from the lexical scope - a private field as a property from `this`.

**Comment by spectranaut** [comment]
@littledan -- thanks for thoughts! in that case, will `f` error when called here:
```
class Outer {
  #x = 42;
  constructor() {
  }

  innerclass() {
    return class extends Outer {
      f() {
	return this.#x;
      }
    }
  }
}
```

**Comment by leobalter** [comment]
https://github.com/tc39/test262/pull/1339/files#diff-7600ad2aeaa7ef2bd4a3bd0266d44563R62

**Comment by littledan** [comment]
@spectranaut Yes, exactly!

**Comment by leobalter** [comment]
I'm assuming this is fixed in the #1339. Please reopen if anything else is missing.
